dehydrated

```diff
@@ -848,13 +848,26 @@ sign_csr() {
   fi
 
   # Finally request certificate from the acme-server and store it in cert-${timestamp}.pem and link from cert.pem
-  echo " + Requesting certificate..."
   csr64="$( <<<"${csr}" "${OPENSSL}" req -config "${OPENSSL_CNF}" -outform DER | urlbase64)"
   if [[ ${API} -eq 1 ]]; then
+    echo " + Requesting certificate..."
     crt64="$(signed_request "${CA_NEW_CERT}" '{"resource": "new-cert", "csr": "'"${csr64}"'"}' | "${OPENSSL}" base64 -e)"
     crt="$( printf -- '-----BEGIN CERTIFICATE-----\n%s\n-----END CERTIFICATE-----\n' "${crt64}" )"
   else
-    result="$(signed_request "${finalize}" '{"csr": "'"${csr64}"'"}' | clean_json | get_json_string_value certificate)"
+    echo " + Finalizing order..."
+    order="$(echo ${finalize} | sed 's,/*[^/]\+/*$,,')"
```

Comment:
This is not always the right URL to get the certificate… I believe you need to get that from the `Location` header returned by the finalize request…

```sh
order="$(signed_request "${finalize}" '{"csr": "'"${csr64}"'"}' 4>&1 | grep -i ^Location: | awk '{print $2}' | tr -d '\r\n')"
```

Note that in that case you don't get the `result` variable to extract the `reqstatus`, maybe there is a simple way to get around that? Else, do the get `order` request as the first thing in the `while` loop in any case to retrieve the status.